# Patch written by fellow Gentoo user ernsteiswuerfel
#
# Please also reffer:
# https://gitlab.gnome.org/World/AbiWord/-/commit/857cd86def49fe8557cfe123830c4d5a61eee732
#
# Closes: https://bugs.gentoo.org/853118
--- a/src/af/xap/xp/xap_Dialog.cpp
+++ b/src/af/xap/xp/xap_Dialog.cpp
@@ -36,8 +36,8 @@

 XAP_Dialog::XAP_Dialog(XAP_DialogFactory * pDlgFactory, XAP_Dialog_Id id,
 		       const char * helpUrl )
-  : m_pApp ( NULL ), m_pDlgFactory ( pDlgFactory ), m_id ( id ), 
-	m_helpUrl(NULL)
+  : m_pApp ( nullptr ), m_pDlgFactory ( pDlgFactory ), m_id ( id ), 
+	m_helpUrl(nullptr)
 {
   m_pApp = pDlgFactory->getApp();

@@ -235,7 +235,7 @@ XAP_Frame *   XAP_Dialog_Modeless::getActiveFrame(void) const
 	// This function returns the frame currently connected to a modeless dialog

 	XAP_Frame * pFrame = m_pApp->getLastFocussedFrame();
-	if(pFrame == (XAP_Frame *) NULL)
+	if(pFrame == (XAP_Frame *) nullptr)
 	{
 		pFrame = m_pApp->getFrame(0);
 	}
@@ -267,7 +267,7 @@ XAP_Dialog_Modeless::BuildWindowName(char * pWindowName, const char * pDialogNam
 // This function constructs and returns the window name of a modeless dialog by
 // concatenating the active frame with the dialog name

-	*pWindowName = (char) NULL;
+	*pWindowName = (char) 0;
 	UT_UTF8String wn = UT_UTF8String(pDialogName);

 	XAP_Frame* pFrame = getActiveFrame();
